Clarify -a switch for FSC and refactor table to markdown (#9495)
dotnet/docs: docs/fsharp/language-reference/compiler-options.md

> 

Reviewer: update ms.date?

|Compiler Option|Description|
|---------------|-----------|
|`-a filename.fs`|Generates a library from the specified file. This option is a short form of `--target:library filename.fs`.|
|`--baseaddress:address`|Specifies the preferred base address at which to load a DLL.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;baseaddress &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/2fdbz5xd.aspx).|

> |`--codepage:id`|Specifies which codepage to use during compilation if the required page is not the current default codepage for the system.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;codepage &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/w0kyekyh.aspx).|

Reviewer: nit: is not -> isn't

not related to your change but this msdn link could be updated to the relative docs link (same for other links used)

should code page be two words? e.g. https://docs.microsoft.com/en-us/windows/desktop/intl/code-pages

|`--consolecolors`|Specifies that errors and warnings use color-coded text on the console.|

> |<code>--crossoptimize[+&#124;-]</code>|Enables or disables cross-module optimizations.|

Reviewer: I believe you could change this to use pipe and backticks now that we're using the markdig engine. same comment for the other rows where the pipe is being encoded.

```suggestion
|`--crossoptimize[+|-]`|Enables or disables cross-module optimizations.|
```

|<code>--checked[+&#124;-]</code>|Enables or disables the generation of overflow checks.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;checked &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/h25wtyxf.aspx).|
|<code>--debug[+&#124;-]</code><br /><br /><code>-g[+&#124;-]</code><br /><br /><code>--debug:[full&#124;pdbonly]</code><br /><br /><code>-g: [full&#124;pdbonly]</code>|Enables or disables the generation of debug information, or specifies the type of debug information to generate. The default is full, which allows attaching to a running program. Choose **pdbonly** to get limited debugging information stored in a pdb (program database) file.<br /><br />Equivalent to the C# compiler option of the same name. For more information, see<br /><br />[&#47;debug &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/8cw0bt21.aspx).|
|`--define:symbol`<br /><br />`-d:symbol`|Defines a symbol for use in conditional compilation.|

> |<code>--deterministic[+&#124;-]</code>|Produce a deterministic assembly (including module version GUID and timestamp).  This cannot be used with wildcard version numbers, and only supports embedded and portable debugging types|

Reviewer: ```suggestion
|<code>--deterministic[+&#124;-]</code>|Produces a deterministic assembly (including module version GUID and timestamp). This option cannot be used with wildcard version numbers, and only supports embedded and portable debugging types|
```

|`--keycontainer:key-container-name`|Specifies a strong name key container.|
|`--keyfile:filename`|Specifies the name of a public key file for signing the generated assembly.|
|`--lib:folder-name`<br /><br />`-I:folder-name`|Specifies a directory to be searched for assemblies that are referenced.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;lib &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/s5bac5fx.aspx).|
|`--linkresource:resource-info`|Links a specified resource to the assembly. The format of resource-info is <code>filename[name[public&#124;private]]</code><br /><br />Linking a single resource with this option is an alternative to embedding an entire resource file with the `--resource` option.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;linkresource &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/xawyf94k.aspx).|

> |`--mlcompatibility`|Ignores warnings that appear when you use features that are designed for compatibility with other versions of ML.|

Reviewer: define what the ML acronym means?

Author: In context, ML is the correct term (since this flag is for compatibility with other ML dialects). ML is short for Hindley **M**i**L**ner (emphasis mine), a family of languages that F# is a part of.

|`--times`|Displays timing information for compilation.|
|`--utf8output`|Enables printing compiler output in the UTF-8 encoding.|
|`--warn:warning-level`|Sets a warning level (0 to 5). The default level is 3. Each warning is given a level based on its severity. Level 5 gives more, but less severe, warnings than level 1.<br /><br />Level 5 warnings are: 21 (recursive use checked at runtime), 22 (`let rec` evaluated out of order), 45 (full abstraction), and 52 (defensive copy). All other warnings are level 2.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;warn &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/13b90fz7.aspx).|

> |`--warnon:warning-number-list`|Enable specific warnings that might be off by default or disabled by another command line option. In F# 3.0, only the 1182 (unused variables) warning is off by default.|

Reviewer: ```suggestion
|`--warnon:warning-number-list`|Enable specific warnings that might be off by default or disabled by another command-line option. In F# 3.0, only the 1182 (unused variables) warning is off by default.|
```

|`--win32manifest:manifest-filename`|Adds a Win32 manifest file to the compilation. This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;win32manifest &#40;C&#35; Compiler Options&#41;](https://msdn.microsoft.com/library/bb545961.aspx).|
|`--win32res:resource-filename`|Adds a Win32 resource file to the compilation.<br /><br />This compiler option is equivalent to the C# compiler option of the same name. For more information, see [&#47;win32res (&#40;C&#35); Compiler Options&#41;](https://msdn.microsoft.com/library/8f2f5x2e.aspx).|

> ## Related Topics

Reviewer: ```suggestion
## Related articles
```
